packages/ripple-nuxt-tide/modules/preview/lib/helpers.js

```diff
@@ -0,0 +1,93 @@
+import { redirect } from '../config/paths'
+import jsonapiParse from 'jsonapi-parse'
+
+async function tidePreview (context, headersConfig) {
+  if (!context.$auth.loggedIn) {
+    return context.redirect(`${redirect.login}?destination=${context.req.url}`)
+  } else {
+    return getPageByPreviewLink(context.app.$tide, context.route.path, context.store.state.tide.siteId, context.route.query.section, headersConfig)
```

Comment:
`context.store.state.tide.siteId` is not the site section id. A page is always under a site and also optionally under a site section. We should get the section id from preview url query string. e.g `/preview/landing_page/d5457459-1735-4b80-8fc6-46e6bdfd74b3/latest?section=224`

Reply:
This getPageByPreviewLink() is based on it's previous tide.js implementation where the $tide module is passed a site id at construction, and the functions like getPageByPath() / getPageByPreviewLink() / getPageByShareLink() all perform use that value to compare to a section ID: 
`pageData.section = pathData.section === site ? null : pathData.section`

In order to move this function into it's own module, keep the same implementation and have access to a site id, I used the value stored in Vuex to pass through to the function. You can see the section id from the query string is also passed as a parameter directly after the site id.

If the `pageData.section` logic needs to change and doesn't require the site ID anymore, perhaps the implementations change for getPageByPath() / getPageByPreviewLink() / getPageByShareLink() ?

Reply:
Hey @alan-cole Not exactly like that. The previous preview is getting the section from route query already, you can just copy the implementation I think: https://github.com/dpc-sdp/ripple/blob/a7b310dc17a3b4d050191fdd2095ebff75e13e49/packages/ripple-nuxt-tide/modules/authenticated-content/lib/middleware.js#L41

Reply:
Sorry, I was referring to these lines in terms of how site and section are used:
https://github.com/dpc-sdp/ripple/blob/master/packages/ripple-nuxt-tide/lib/core/tide.js#L419
https://github.com/dpc-sdp/ripple/blob/master/packages/ripple-nuxt-tide/lib/core/tide.js#L442

In my above code, I am using site section from query..

return getPageByPreviewLink(context.app.$tide, context.route.path, context.store.state.tide.siteId, **context.route.query.section**, headersConfig)

Reply:
@alan-cole ahh, my bad, looked at wrong argument. It's correct then. One more thing, do you need to check if those context data are available? As chained object property can cause errors if the parent is undefined.

Reply:
Yeah, that would be safest. I'll add some checks to prevent any breakages.

Reply:
My understanding is var `route.query` is always available, and if section isn't defined then it will return `undefined`.

Either

`const section = (context.route.query && context.route.query.section) ? context.route.query.section : null`

or

`const section = context.route.query.section`

will result in a falsy value and shouldn't make a difference to the execution and both should cause `pageData.section` to be null.

Similar to `context.store.state.tide.siteId`. I believe it's unlikely to be unset in practice.

Anyway, I've placed some checks just in case.

Maybe for a future improvement, as IE is the only browser left that doesn't support optional chaining, perhaps we could look at using:
https://babeljs.io/docs/en/next/babel-plugin-proposal-optional-chaining
to help clean up the codebase?